lib/ts/TextView.cc

```diff
@@ -50,19 +50,18 @@ ts::memcmp(TextView const &lhs, TextView const &rhs)
 }
 
 int
-ts::strcasecmp(TextView lhs, TextView rhs)
+strcasecmp(const std::string_view &lhs, const std::string_view &rhs)
 {
-  while (lhs && rhs) {
-    char l = tolower(*lhs);
-    char r = tolower(*rhs);
-    if (l < r) {
-      return -1;
-    } else if (r < l) {
-      return 1;
+  size_t len = std::min(lhs.size(), rhs.size());
+  int zret   = strncasecmp(lhs.data(), rhs.data(), len);
```

Comment:
You could do the length comparison before doing the strncasecmp.  But that is a nit.  Looks like a good replacement to use the systemcall rather than the local implementation.

Reply:
Sadly not. Because it's a "cmp" function it  needs to return lexicographic order and not just difference. For instance, "zoo" vs. "alpha" must return 1, but "axe" vs. "zaphod" must return -1.